internal/app/streamer_test.go

```diff
@@ -133,3 +218,29 @@ func (bp *basinProviderMock) LatestDeals(context.Context, string, string, uint32
 func (bp *basinProviderMock) Reconnect() error {
 	return nil
 }
+
+func (bp *basinProviderMock) Upload(
```

Comment:
the mock impl, makes a copy of the file that was uploaded. to test, we will import this file into a db and make assertions